MOLFCMClient/MOLFCMClient.h

```diff
@@ -42,23 +48,39 @@ typedef void (^MOLFCMAcknowledgeErrorHandler)(NSDictionary *, NSError *);
  *  The designated initializer.
  *
  *  @param FCMToken             FCM Token to identify and authenticate the client
- *  @param sessionConfiguration The desired NSURLSessionConfiguration, can be nil
- *  @param messageHandler      The block to be called for every message received
+ *  @param host                 Optional, host (i.e. https://localhost:8080)
+ *  @param connectDelayMax      Optional, max delay (seconds) when calling connect
+ *  @param backoffMax           Optional, max backoff (seconds) when the connection is interrupted
+ *  @param fatalCodes           Optional, do not attempt to reconnect if a fatal code is returned
+ *  @param sessionConfiguration Optional, the desired NSURLSessionConfiguration
+ *  @param messageHandler       The block to be called for every message received
  *
+ *  @note If the host argument is nil, https://fcm-stream.googleapis.com will be used.
+ *  @note If the connectDelayMax argument is 0, a default value of 10 will be used.
+ *  @note If the backoffMax argument is 0, a default value of 900 will be used.
+ *  @note If the fatalCodes argument is nil, @[@302, @400, @403] will be used.
  *  @note If the sessionConfiguration argument is nil, defaultSessionConfiguration will be used.
  *
  *  @return An initialized MOLFCMClient object
  */
 - (instancetype)initWithFCMToken:(NSString *)FCMToken
+                            host:(NSString *)host
+                 connectDelayMax:(uint32_t)connectDelayMax
+                      backoffMax:(uint32_t)backoffMax
+                      fatalCodes:(NSArray<NSNumber *> *)fatalCodes
             sessionConfiguration:(NSURLSessionConfiguration *)sessionConfiguration
-                  messageHandler:(MOLFCMMessageHandler)messageHandler
-                     NS_DESIGNATED_INITIALIZER;
+                  messageHandler:(MOLFCMMessageHandler)messageHandler NS_DESIGNATED_INITIALIZER;
```

Comment:
If possible the initializer you're replacing should still exist and just call the new one

Reply:
sure thing